PX4/failsafe-overhaul.md

```diff
@@ -0,0 +1,136 @@
+- Feature Name: failsafe_overhaul
+- Start Date: 2022-07-19
+- RFC PR: [dronecode/rfc#0000](https://github.com/dronecode/rfc/pull/0000)
+
+# Summary
+[summary]: #summary
+
+The current implementation of failsafe actions in PX4 is very rigid and limited. It is impossible to combine failsafe conditions, trigger multiple actions, or set progressively more aggressive actions (currently hardcoded fore battery voltage monitoring). It is also very difficult to extend the failsafe system to respond to user specific conditions because it is tightly tied into `commander`. Generalizing failsafe configuration to a standard framework with variable conditions and actions allows more flexibility. Additionally, more conditions and actions should be added.
```

Comment:
Typo "for"
```suggestion
The current implementation of failsafe actions in PX4 is very rigid and limited.
It is impossible to combine failsafe conditions, trigger multiple actions, or set progressively more aggressive actions (currently hardcoded for battery voltage monitoring).
It is also very difficult to extend the failsafe system to respond to user specific conditions because it is tightly tied into `commander`. Generalizing failsafe configuration to a standard framework with variable conditions and actions allows more flexibility. Additionally, more conditions and actions should be added.
```